src/meta/src/manager/catalog/mod.rs

```diff
@@ -324,6 +335,24 @@ where
         }
     }
 
+    pub async fn create_connection(
+        &self,
+        service_name: &str,
+        connection: Connection,
+    ) -> MetaResult<()> {
+        let core = &mut self.core.lock().await.connection;
+
+        let conn_id = connection.id;
+        let mut connections = BTreeMapTransaction::new(&mut core.connections);
+        // if there is a connection with the same service name, just overwrite it
+        connections.insert(conn_id, connection);
+        commit_meta!(self, connections)?;
```

Comment:
There's some async points in `commit_meta`. As explained in `https://github.com/risingwavelabs/risingwave/pull/8145`, It is possible that the in-memory cache and the meta store are inconsistent if the caller get killed. So I'm affraid that we need to run it in `ddl_controller`.

Reply:
Fixed